Arriba/Arriba/Model/Query/JoinQuery.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using Arriba.Model.Correctors;
+using Arriba.Model.Expressions;
+
+namespace Arriba.Model.Query
+{
+    /// <summary>
+    ///  IQuery wrapper which implements Joins.
+    ///  During correction, the JoinCorrector corrects additional joined queries
+    ///  and then replaces terms in the primary query by evaluating the referenced join
+    ///  and getting all results for the specified column.
+    ///  
+    ///  For example, if we have:
+    ///    Q = [GroupName]:Nice AND [Members]::#Q1[Alias]
+    ///    Q1 = [TeamName]:T1
+    ///    
+    ///  Then correcting 'Q' will cause the JoinCorrector to run Q1, get all
+    ///  values for the 'Alias' column, split them, and replace [Members]::#Q1[Alias]
+    ///  with [Members]:: IN(alias1, alias2, alias3, ...).
+    /// </summary>
+    public class JoinQuery<T> : IQuery<T>
```

Comment:
The reason why I specifically suggested it to be class JoinQuery : SelectQuery is so that a JoinQuery could be nested inside of a JoinQuery.  Without that inheritance, you can't.